base/inference.jl

```diff
@@ -1873,7 +1873,7 @@ function limit_tuple_type_n(@nospecialize(t), lim::Int)
     p = t.parameters
     n = length(p)
     if n > lim
-        tail = reduce(typejoin, Bottom, Any[p[lim:(n-1)]..., unwrapva(p[n])])
+        tail = reduce(promote_join, Bottom, Any[p[lim:(n-1)]..., unwrapva(p[n])])
```

Comment:
This should not be changed in inference.

Reply:
Right. I thought this was needed to change the promotion of tuples, but clearly I've been overzealous. I've reverted it.